From a3b84f1777bd75bdc5914f4ce5f52c55e723cd9a Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Sat, 3 Jun 2017 21:23:52 +0200
Subject: [PATCH] sh/ptrace: Define pt_{dsp,}regs uapi_pt_{dsp,}regs on !GLIBC
 systems

Fixes a pt_{dsp,}regs redefinition when building with the musl C library
on SuperH.

Inspired by
http://git.yoctoproject.org/clean/cgit.cgi/poky/plain/meta/recipes-devtools/gdb/gdb/0004-ppc-ptrace-Define-pt_regs-uapi_pt_regs-on-GLIBC-syst.patch,
adapted for SuperH.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
[Rebase	on gdb 8.0]
Signed-off-by: Romain Naour <romain.naour@gmail.com>
---
 gdbserver/linux-sh-low.cc | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/gdbserver/linux-sh-low.cc b/gdbserver/linux-sh-low.cc
index 966bdeb8ba7..1e3ce53b4d5 100644
--- a/gdbserver/linux-sh-low.cc
+++ b/gdbserver/linux-sh-low.cc
@@ -76,7 +76,15 @@ extern const struct target_desc *tdesc_sh;
 #include <sys/reg.h>
 #endif
 
+#if !defined(__GLIBC__)
+# define pt_regs uapi_pt_regs
+# define pt_dspregs uapi_pt_dspregs
+#endif
 #include <asm/ptrace.h>
+#if !defined(__GLIBC__)
+# undef pt_regs
+# undef pt_dspregs
+#endif
 
 #define sh_num_regs 41
 
-- 
2.37.1

